modules/api_gateway/main.tf

```diff
@@ -105,26 +97,34 @@ resource "aws_iam_role_policy" "invocation_policy" {
       {
         "Action" = "lambda:InvokeFunction",
         "Effect" = "Allow",
-        "Resource" = distinct([
-          # use wildcard for versioned lambdas to avoid
-          # revoking permissions on previous version during deploy
-          for lambda_key, value in var.lambdas : replace(value.function_arn, "/:\\d+$/", ":*")
-        ])
+        "Resource" = [for e in distinct(
+          concat([
+            # use wildcard for versioned lambdas to avoid
+            # revoking permissions on previous version during deploy
+            for key, value in var.authorizers : replace(value.function_arn, "/:\\d+$/", ":*") if lookup(value, "function_arn", "") != ""
```

Comment:
Does AWS give you an error if the resource is in another account? Is that why this is needed in the first place? I'm just curious what the justification is for this PR. I'm assuming AWS is rejecting the change somewhere, it's just not obvious to me what the issue is.

Reply:
> Does AWS give you an error if the resource is in another account?

Surprisingly no. It appears that there is some _magic_ under the hood when it comes to cross-account authorizers.